rpcs3/Emu/RSX/Common/ProgramStateCache.h

```diff
@@ -478,48 +490,40 @@ class program_state_cache
 		}
 		else
 		{
-			reader_lock lock(m_decompiler_mutex);
-
-			auto vertex_program_found = std::find_if(m_decompile_queue.begin(), m_decompile_queue.end(), [&](const auto& V)
-			{
-				if (V.is_fp) return false;
-				return program_hash_util::vertex_program_compare()(V.vp, vertexShader);
-			});
+			bool add_vertex_program = true, add_fragment_program = true;
 
-			auto fragment_program_found = std::find_if(m_decompile_queue.begin(), m_decompile_queue.end(), [&](const auto& F)
+			for (int i = 0; i < m_decompile_queue.size(); ++i)
 			{
-				if (!F.is_fp) return false;
-				return program_hash_util::fragment_program_compare()(F.fp, fragmentShader);
-			});
+				auto& P = m_decompile_queue[i];
```

Comment:
This should have been a properly named variable as this isn't a lambda, and also should be const auto. None of that matters right now since lf_queue does not work with indices anyway and is what we want.